### PR TITLE
chore: add `check_code_format.yml`

### PR DESCRIPTION
**Pull request by vil02**
This PR adds a workflow, which checks the used code formatting (and makes the necessary changes in the currently existing files).

As suggested by @Ramy-Badr-Ahmed in #13, the new workflow uses [`fprettify`](https://github.com/fortran-lang/fprettify). I could not force it to fail, if there are some changes necessary, so I used a _fail if needs reformatting_ approach as in TheAlgorithms/Nim#19.

Beside the indentation size being set to `4`, I used the default setting of the `fprettify`. Please let me know if you prefer other settings.

The _failing part_ works [logs](https://github.com/TheAlgorithms/Fortran/actions/runs/11021300048/job/30608062137) for 8663d150dc0a7c6c9fbc49e74405cd7bdd508c43.

### CODE REVIEW

**Comment by Ramy-Badr-Ahmed** [comment]
> Beside the indentation size being set to `4`, I used the default setting of the `fprettify`. Please let me know if you prefer other settings.
> 
> The _failing part_ works [logs](https://github.com/TheAlgorithms/Fortran/actions/runs/11021300048/job/30608062137) for [8663d15](https://github.com/TheAlgorithms/Fortran/commit/8663d150dc0a7c6c9fbc49e74405cd7bdd508c43).

Adding the debug output `-d` to `fprettify --indent 4 -d --recursive .` can help identifiy the styling mismatches by contributors.

Here's an example output: [logs](https://github.com/Ramy-Badr-Ahmed/Fortran-DSA/actions/runs/11022747314/job/30612715973)

**Comment by vil02** [comment]
> > Beside the indentation size being set to `4`, I used the default setting of the `fprettify`. Please let me know if you prefer other settings.
> > The _failing part_ works [logs](https://github.com/TheAlgorithms/Fortran/actions/runs/11021300048/job/30608062137) for [8663d15](https://github.com/TheAlgorithms/Fortran/commit/8663d150dc0a7c6c9fbc49e74405cd7bdd508c43).
> 
> Adding the debug output `-d` to `fprettify --indent 4 -d --recursive .` can help identifiy the styling mismatches by contributors.
> 
> Here's an example output: [logs](https://github.com/Ramy-Badr-Ahmed/Fortran-DSA/actions/runs/11022747314/job/30612715973)

Strange... the output does not appear in the actions logs:
- https://github.com/TheAlgorithms/Fortran/actions/runs/11040556555/job/30668810319
- https://github.com/TheAlgorithms/Fortran/actions/runs/11040442388/job/30668440497
- https://github.com/TheAlgorithms/Fortran/actions/runs/11040275300/job/30667876416

EDIT: of course the output is empty... the files are already reformatted. I will use `git diff`.

**Comment by Ramy-Badr-Ahmed** [comment]
> Strange... the output does not appear in the actions logs:
>     * https://github.com/TheAlgorithms/Fortran/actions/runs/11040556555/job/30668810319
>     * https://github.com/TheAlgorithms/Fortran/actions/runs/11040442388/job/30668440497
>     * https://github.com/TheAlgorithms/Fortran/actions/runs/11040275300/job/30667876416

I see. `-d` flag shows differences, no changes made.

I tried it out and it would work with `git diff` in the `failure` step:

```
      - name: Format Fortran files
        run: |          
          fprettify -i 4 --recursive .
      - name: Fail if needs reformatting      
        run: |  
          if [[ $(git status --porcelain) ]]; then            
            echo "please reformat/fprettify these files:"            
            git diff
            exit 1
          else
            echo "All files are properly formatted."
          fi   
``` 

My logs: https://github.com/Ramy-Badr-Ahmed/Fortran-DSA/actions/runs/11043111068/job/30676793615

Let's have another try 🙂

File: .github/workflows/check_code_format.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+      - name: Display fprettify version
+        run: fprettify --version
+
+      - name: Format code
+        run: |
+          git clean -f -x -d
+          fprettify --indent 4 --recursive .
+
+      - name: Fail if needs reformatting
```
Comment (Ramy-Badr-Ahmed) [inline .github/workflows/check_code_format.yml:29]:
```
      - name: Fail if needs reformatting      
        run: |  
          if [[ $(git status --porcelain) ]]; then            
            echo "please reformat/fprettify these files:"            
            git diff
            exit 1
          else
            echo "All files are properly formatted."
          fi
```

Comment (vil02) [reply .github/workflows/check_code_format.yml:29]: It works now, cf. https://github.com/TheAlgorithms/Fortran/actions/runs/11045781263/job/30683988597

**Comment by vil02** [comment]
So as mentioned above: the output was empty, because _obviously_ the files were already reformatted. I used `git diff`.

Now it works as expected, cf. the failing case: https://github.com/TheAlgorithms/Fortran/actions/runs/11045781263/job/30683988597

File: .github/workflows/check_code_format.yml (hunk 1 above)
Comment (Ramy-Badr-Ahmed) [reply .github/workflows/check_code_format.yml:29]: Perfect 👍

**Review by Ramy-Badr-Ahmed — COMMENTED** [review]
Yes, it's now very informative 👍

**Review by Ramy-Badr-Ahmed — APPROVED** [review]

**Review by SatinWukerORIG — APPROVED** [review]
👍 !

**Pull request merged**